File: common/src/main/java/com/asiainfo/ocdp/common/router/LocalOrBanlanceRouter.java
package com.asiainfo.ocdp.common.router;

/**
 * <pre>
 *
 * Created by zhenqin.
 * User: zhenqin
 * Date: 16/10/9
 * Time: 12:46
 * Verdor: NowledgeData
 * To change this template use File | Settings | File Templates.
 *
 * </pre>
 *
 * @author zhenqin
 */
public class LocalOrBanlanceRouter extends BanlanceRouter {


    public LocalOrBanlanceRouter(String cacheManagers) {
        super(cacheManagers);
    }


    @Override
    public String getProxyHost(String host) {
        String codisHost = this.PROXY_HOST_MAP.get(host);
        if (codisHost == null) {
            return super.getProxyHost(host);
        }
        return codisHost;
    }
}

File: core/src/main/scala/com/asiainfo/ocdp/stream/common/LocalOrBanlanceRouter.scala
package com.asiainfo.ocdp.stream.common

import java.util

import redis.clients.jedis.JedisPool

/**
  *
  *
  * <pre>
  * User: liuyu
  * Date: 2016/10/9
  * Time: 10:31
  * </pre>
  *
  * @author liuyu
  */
class LocalOrBanlanceRouter  extends BanlanceRouter {

	def this(cacheManager: String) {
		this()
		this.cacheManager=cacheManager
		val cacheManagers: Array[String] = this.cacheManager.split(",")
		for (manager <- cacheManagers) {
			val split: Array[String] = manager.split(":")
			val maybeList = new util.LinkedList[String]()
			if(hostMap.contains(split(0))){
				maybeList.addAll(hostMap.get(split(0)).get)
			}
			maybeList.add(manager)
			hostMap.put(split(0),maybeList)
		}
	}

	override def proxyHost(host: String): JedisPool = {
		var jedisPool:JedisPool = null

		jedisPool = new LocalRouter(cacheManager).proxyHost(host)

		if (jedisPool == null ) {
			jedisPool = super.proxyHost(host)
		}

		jedisPool
	}
}
